src/Microsoft.AspNetCore.SystemWebAdapters.CoreServices/Authentication/IAuthenticationResultFactory.cs

```diff
@@ -10,7 +10,7 @@ namespace Microsoft.AspNetCore.SystemWebAdapters.Authentication;
 /// Interface for creating authentication results from HTTP responses returned
 /// by a remote app in remote authentication scenarios.
 /// </summary>
-internal interface IAuthenticationResultFactory
+public interface IAuthenticationResultFactory
```

Comment:
We'll probably need to do an API review for this

Reply:
Agreed

Reply:
I'm not seeing a need to expose this for the serialization - am I missing something?

Reply:
This is the interface whose implementation (`RemoteAppAuthenticationResultFactory`) uses the claims serialization. Making this public doesn't relate to the effort to make the serialization logic extensible directly, but it relates to the scenario of needing to customize how auth decisions are made in the ASP.NET Core app.

The advantage to this interface being public is that if, for some reason, a user's authentication solution was such that just changing how claims are serialized/deserialized wasn't enough to support getting identity from their ASP.NET app's authentication endpoint, they could implement this interface and have even broader control over how the response from the ASP.NET app is handled.

Reply:
After thinking about this more, I'm going to leave this as internal for now since there's not an immediate request for it to be public and it will be a simple one-line change to make it public in the future, if needed. I'll follow-up with API review folks on the new IClaimsSerializer interface.